app/src/main/java/org/mozilla/focus/fragment/BrowserFragment.kt

```diff
@@ -135,7 +136,9 @@ class BrowserFragment : IWebViewLifecycleFragment(),
                 reload()
             }
             NavigationEvent.PIN_SITE -> {
-                // TODO: Add to pinned sites
+                if (webview.getUrl() != null) {
```

Comment:
If you want to avoid the `!!` after the null check, you need to smart-cast. Smart-cast can only happen if the value is guaranteed not to change, which happens if you assign it to an immutable value:
```kotlin
val url = webview.getUrl()
if (url != null) {
    HomeTiles...
}
```

Or:

```kotlin
webview.getUrl()?.let { url -> // or use `it` directly
    HomeTiles...
}
```

Note that the `?` is really important: iirc, you can call `let` without it (which is really weird and not a good thing so I generally don't use let).

Reply:
thanks! that's exactly what I was looking for :)